client/me/purchases/payment-methods/existing-credit-card.tsx

```diff
@@ -0,0 +1,432 @@
+				const previousData = queryClient.getQueryData( queryKey );
+				// Optimistically update the fields
+				queryClient.setQueryData( queryKey, {
+					tax_postal_code: onMutateInputValues.tax_postal_code,
+					tax_country_code: onMutateInputValues.tax_country_code,
+					is_tax_info_set: true,
+				} );
+				return { previousData };
+			},
+			onError: ( error, _, context ) => {
+				// Restore previous fields
+				queryClient.setQueryData( queryKey, context?.previousData );
+				setUpdateError( ( error as Error ).message );
+			},
+			onSuccess: ( onSuccessInputValues: TaxInfo ) => {
+				queryClient.setQueryData( queryKey, {
+					tax_postal_code: onSuccessInputValues.tax_postal_code,
+					tax_country_code: onSuccessInputValues.tax_country_code,
+					is_tax_info_set: true,
+				} );
+				closeDialog();
+			},
+		}
+	);
+
+	const openDialog = useCallback( () => {
+		taxInfoFromServer && setInputValues( taxInfoFromServer );
+		setIsDialogVisible( true );
+	}, [ taxInfoFromServer ] );
+
+	useEffect( () => {
+		if ( ! taxInfoFromServer?.tax_country_code ) {
+			return;
+		}
+		setInputValues( {
+			tax_postal_code: taxInfoFromServer?.tax_postal_code ?? '',
+			tax_country_code: taxInfoFromServer?.tax_country_code ?? '',
+		} );
+	}, [ taxInfoFromServer?.tax_country_code, taxInfoFromServer?.tax_postal_code ] );
+
+	const updateTaxInfo = useCallback( () => {
+		mutation.mutate( inputValues );
+	}, [ mutation, inputValues ] );
+
+	const onChangeCountryCode = ( e: { target: { value: string } } ) => {
+		setInputValues( { ...inputValues, tax_country_code: e.target.value } );
+	};
+
+	const onChangePostalCode = ( e: { target: { value: string } } ) => {
+		setInputValues( { ...inputValues, tax_postal_code: e.target.value } );
+	};
+
+	const formRender = (
+		<form>
+			<div className="contact-fields payment-methods__tax-fields">
+				<RenderEditFormFields
+					postalCodeValue={ inputValues.tax_postal_code }
+					countryCodeValue={ inputValues.tax_country_code }
+					onChangePostalCode={ onChangePostalCode }
+					onChangeCountryCode={ onChangeCountryCode }
+				/>
+			</div>
+		</form>
+	);
+
+	/* translators: %s is the last 4 digits of the credit card number */
+	const maskedCardDetails = sprintf( _x( '**** %s', 'Masked credit card number' ), last4 );
+
+	const taxInfoDisplay = joinNonEmptyValues(
+		', ',
+		taxInfoFromServer?.tax_postal_code,
+		taxInfoFromServer?.tax_country_code
+	);
+
+	return (
+		<Fragment>
+			<div>
+				<CardHolderName>{ cardholderName }</CardHolderName>
+				<CardDetails>{ maskedCardDetails }</CardDetails>
+				<span>{ `${ __( 'Expiry:' ) } ${ formatDate( cardExpiry ) }` }</span>
+				{ taxInfoDisplay ? (
+					<span className="existing-credit-card__tax-info-display">
+						<span className="existing-credit-card__tax-info-postal-country">
+							{ taxInfoDisplay }
+						</span>
+						<PaymentMethodEditButton
+							onClick={ openDialog }
+							buttonTextContent={ __( 'Edit' ) }
```

Comment:
IMO we should not show the Edit button here. I think it's confusing for several reasons:

- It might imply that you can edit the other information about the card.
- Why can't we edit the tax info on the payment method list (or even see it)?
- How likely is it that we will need this?

Plus, we can always add this later.

Reply:
Agreed. I've removed this Edit instance of it. Since the component is being used for the Missing Billing Info button, it's trivial to re-add it later if needed.